### PR TITLE
chore: Split Stateful Single Stack to Multi Stack

### LINKED ISSUE #188: Rename/Split Shared Resources in Stateful Deployment Stack
In the deployment of the stateful pipeline, all resources are currently in 1 stack. We should split or rename this to become a shared resource stack as there might be multiple service requiring its own stateful resource. Isolating the shared resource from other service resources could make easier deployment testing and breaking it wouldn't affect other services.

### PR DESCRIPTION
**Pull request by williamputraintan**
- mainly refactoring that shared resource are all in one stack and any other µ-app that require stateful be on different stack (preventing new µ-app feature to step on SharedStack's toe)
- wrapping multiple stateful stacks in a class called `StatefulStackCollection` when defined in stage
- refactoring on typescript naming convention

Example on `cdk ls` command after refactoring:

```bash
yarn cdk-stateful ls

OrcaBusStatefulPipeline
OrcaBusStatefulPipeline/BetaDeployment/IcaEventPipeStack (BetaDeployment-IcaEventPipeStack)
OrcaBusStatefulPipeline/BetaDeployment/SharedStack (BetaDeployment-SharedStack)
OrcaBusStatefulPipeline/BetaDeployment/TokenServiceStack (BetaDeployment-TokenServiceStack)
...
```

resolves #188

### CODE REVIEW

File: README.md
Comment (williamputraintan) [inline README.md:124]: I have been inconsistent with naming convention, but I think we could follow what AWS suggestion here.

Folder naming convention is not mentioned, but in general `kebab-case` is more common. For example, AWS uses `kebab-case` in their CDK library (`import { IVpc } from 'aws-cdk-lib/aws-ec2';`)

Comment (williamputraintan) [reply README.md:124]: will refactor on the stateless stack after this PR along with removing the parent (of the µ-app) stack

https://github.com/umccr/orcabus/pull/157#issuecomment-2000650031

**Comment by williamputraintan** [comment]
This might need a redeployment of resources here as stack id will change.

We could also roll this to all of our account env (dev, stg, prod) once this is approved

**Comment by victorskl** [comment]
.. will review today

**Comment by victorskl** [comment]
reviewing...

File: README.md
Comment (victorskl) [reply README.md:124]: Sound good. I support this.

File: docs/developer/SHARED_RESOURCES.md
Comment (victorskl) [reply docs/developer/SHARED_RESOURCES.md:2]: Feel, 😬 a little bit of double efforts/handling here, yes. 

Up to the team with favour the change/remove; ie stick the README closer to the stack/resource that is deployed. At least for these early days taking off phases. We tend to change .. a lot, frequently.

File: lib/workload/components/vpc/index.ts
Comment (victorskl) [inline lib/workload/components/vpc/index.ts:1]: We can also remove this VPC lookup function i.e. rather promote to use the `vpcProps` pass down from the `constants.ts` and, reconstruct at the stack level. Like the way it does in Token Service stack.

What chu think, Will? Leaving it is fine, too.

File: lib/workload/stateful/stacks/shared/constructs/database/index.ts
Comment (victorskl) [inline lib/workload/stateful/stacks/shared/constructs/database/index.ts:115]: Thank you for cleaning this id thingy up.! 😬

File: lib/workload/stateful/stacks/shared/constructs/schemaregistry/index.ts
Comment (victorskl) [inline lib/workload/stateful/stacks/shared/constructs/schemaregistry/index.ts:1]: Bye, bye, my dear "component". Hehe. J/K 😬

File: lib/workload/stateful/stacks/token-service/deploy/stack.ts
Comment (victorskl) [inline lib/workload/stateful/stacks/token-service/deploy/stack.ts:39]: Ah. You want the function one. Please check with the Flo! 😃 Think, he prefer the other way, `vpcProps` pass down. Leave it with you two.

File: lib/workload/stateful/statefulStackCollectionClass.ts
Comment (victorskl) [inline lib/workload/stateful/statefulStackCollectionClass.ts:14]: Oh, the "collection" class. I like this naming.! Good one, Will. 👍

**Review by victorskl — APPROVED** [review]
👍 from me.!

File: lib/workload/components/vpc/index.ts
Comment (williamputraintan) [reply lib/workload/components/vpc/index.ts:1]: Agree on lookup at µ-stack level rather than passing `IVpc` as a stack props. I feel it is more isolated and it is simpler to get onboard when dev the µ-app rather depending on resource as the props.

Comment (williamputraintan) [reply lib/workload/components/vpc/index.ts:1]: I just thought this function is just a quick single-line wrapper of common resource lookup

File: lib/workload/stateful/statefulStackCollectionClass.ts
Comment (williamputraintan) [reply lib/workload/stateful/statefulStackCollectionClass.ts:14]: Initially was set to "manager" class, but I think we had enough managers in OrcaBus 🤔

File: lib/workload/stateful/stacks/token-service/deploy/stack.ts
Comment (reisingerf) [reply lib/workload/stateful/stacks/token-service/deploy/stack.ts:39]: Hm,... does the function add much? 
It's a bit easier for me without... first I don't need to know about any outside function and second it's just a lookup like I'd need to do for any other (provided) resource.
Also, `getVpc()` hides away configuration details that otherwise would be in the usual, expected spot (config constants), right?

**Review by reisingerf — APPROVED** [review]
Looks nice and neat! Go for it!

File: lib/workload/stateful/stacks/token-service/deploy/stack.ts
Comment (victorskl) [reply lib/workload/stateful/stacks/token-service/deploy/stack.ts:39]: Yup. Since we change to independent "stack" concept, the props password become more make sense.

Think, Will agree (somewhere ^ there comment; deprecating this function lookup all together). Perhaps, Will could help make necessary changes and, use `vpcProps` pass down. I reckon, that should work... But, leaving the final call to Will.

Comment (williamputraintan) [reply lib/workload/stateful/stacks/token-service/deploy/stack.ts:39]: Will remove it! It will be easier for native lookup rather than understanding a custom function.

Comment (victorskl) [reply lib/workload/stateful/stacks/token-service/deploy/stack.ts:39]: Yay! 🎉

File: lib/workload/stateful/statefulStackCollectionClass.ts
Comment (williamputraintan) [inline lib/workload/stateful/statefulStackCollectionClass.ts:41]: added a few more initial tagging template here for each µ-stack @victorskl @reisingerf

Comment (reisingerf) [reply lib/workload/stateful/statefulStackCollectionClass.ts:41]: Nice!

**Review by reisingerf — APPROVED** [review]

**Pull request merged**
